### PR TITLE
gh-108303: Remove the non-test `Lib/test/reperf.py`

### PR DESCRIPTION
**Pull request by sobolevn**
Looks like this file was not touched for at least 12 years: https://github.com/python/cpython/commits/main/Lib/test/reperf.py

It does not seem useful. If it is, I can instead move it to `Tool/scripts`.

Refs https://github.com/python/cpython/pull/114354

* Issue: gh-108303

### CODE REVIEW

**Review by gpshead — APPROVED** [review]

**Review by gpshead — DISMISSED** [review]
apparently test_embed.py runs this code?

**Comment by sobolevn** [comment]
Ok, looks like it was used recently in https://github.com/python/cpython/pull/110769 by @ambv 
I think that this has nothing to do with `reperf`, it just needs any rather complex module.

I propose to:
- replace `test_embed`'s code to generate the module from code, so it would have no depedencies
- still remove this module, we are cleaning things up, aren't we? :)

**Comment by sobolevn** [comment]
I am now using another file instead of `reperf`, it is already used in `test_embed` and is useful for this test as well: it prints the information, it has some imports, it does some checks.

So, we can remove the unneeded `reperf` 🎉

**Review by encukou — APPROVED** [review]
LGTM, but I'd rather not backport the removal.

**Comment by sobolevn** [comment]
@encukou looks like this PR https://github.com/python/cpython/pull/113601 influenced the CI. I fix tests, so now both `./python.exe -m test test_embed` and `./python.exe Lib/test/_test_embed_structseq.py` pass:

```
» ./python.exe Lib/test/_test_embed_structseq.py            
..
----------------------------------------------------------------------
Ran 2 tests in 0.000s

OK
Tests passed
```

Please, take another look :)

File: Lib/test/_test_embed_structseq.py [hunk 1]
```diff
@@ -48,7 +48,14 @@ def test_sys_funcs(self):
...
 try:
-    unittest.main()
+    unittest.main(
+        module=(
+            '__main__'
+            if __name__ == '__main__'
+            # Avoiding a circular import:
+            else sys.modules['test._test_embed_structseq']
+        )
+    )
```
Comment (encukou) [inline Lib/test/_test_embed_structseq.py:58]: Hmm, if `unittest.main` takes module names, would `unittest.main(__name__)` work here?

```suggestion
    # This should run as a script even if it's not `__main__`.
    unittest.main(__name__)
```

Comment (sobolevn) [reply Lib/test/_test_embed_structseq.py:58]: No: 

```
» ./python.exe -m test test_embed -m test_presite           
Using random seed: 989874951
0:00:00 load avg: 2.36 Run 1 test sequentially in a single process
0:00:00 load avg: 2.36 [1/1] test_embed
test test_embed failed -- Traceback (most recent call last):
  File "/Users/sobolev/Desktop/cpython2/Lib/test/test_embed.py", line 1987, in test_presite
    self.assertIn("Tests passed", out)
    ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
AssertionError: 'Tests passed' not found in 'pre-site import failed:\nTraceback (most recent call last):\n  File "/Users/sobolev/Desktop/cpython2/Lib/test/_test_embed_structseq.py", line 51, in <module>\n    unittest.main(\n    ~~~~~~~~~~~~~^\n        __name__\n        ^^^^^^^^\n    )\n    ^\n  File "/Users/sobolev/Desktop/cpython2/Lib/unittest/main.py", line 74, in __init__\n    self.module = getattr(self.module, part)\n                  ~~~~~~~^^^^^^^^^^^^^^^^^^^\nAttributeError: cannot access submodule \'_test_embed_structseq\' of module \'test\' (most likely due to a circular import)\nunique-python-message'

test_embed failed (1 failure)

== Tests result: FAILURE ==

1 test failed:
    test_embed

Total duration: 102 ms
Total tests: run=1 (filtered) failures=1
Total test files: run=1/1 (filtered) failed=1
Result: FAILURE
```

**Comment by encukou** [comment]
OK. Thanks!

**Pull request merged**